test/suites/greenboot.robot

```diff
@@ -0,0 +1,109 @@
+*** Settings ***
+Documentation       Tests related to greenboot
+Resource            ../resources/common.resource
+Resource            ../resources/systemd.resource
+Resource            ../resources/ostree-health.resource
+Resource            ../resources/microshift-config.resource
+Library    SSHLibrary
+Suite Setup         Setup Suite 
+Suite Teardown      Teardown Suite
+
+
+*** Variables ***
+${LOCAL_CONFIG_FILE}                  ../docs/config/busybox_running_check.sh
+${REMOTE_CONFIG_FILE}                 /etc/greenboot/check/required.d/50_busybox_running_check.sh
+${GREENBOOT_CONFIG_FILE}              /etc/greenboot/greenboot.conf
+${GREENBOOT_CONFIG_CONTENT}           MICROSHIFT_WAIT_TIMEOUT_SEC=180
+${MANIFEST_SOURCE_DIR}                ./assets/kustomizations/greenboot/
+${MANIFEST_DIR}                       /etc/microshift/manifests
+${MICROSHIFT_CONFIG}                  /etc/microshift/config.yaml
+*** Test Cases ***
+
+# user-workload-health
+
+GreenbootHealth Run with User Workload
+    [Documentation]    check
+    Upload User Config
+    Microshift Cleanup And Start
+    [Teardown]    Run Keywords
+    ...    Cleanup User Config
+    ...    Microshift Cleanup And Start
+
+GreenbootHealth Microshift Service Failure
+    [Documentation]    Microshift Service failure
+    Microshift Disrupt Service
+    Run Keyword And Expect Error    	0 != 1
+    ...    Microshift Cleanup And Start
+    [Teardown]    Run Keywords
+    ...    Microshift Restore Service
+    ...    Microshift Cleanup And Start
+
+GreenbootHealth Microshift Pod Failure
+    [Documentation]    Microshift Pod Network failure
+    Microshift Disrupt Pod Network
+    Restart MicroShift
+    Run Keyword And Expect Error    	0 != 1
+    ...    Greenboot Healthy
+    [Teardown]    Run Keywords
+    ...    Clear MicroShift Config
+    ...    Microshift Cleanup And Start
+
+
+*** Keywords ***
+Setup Suite
+    [Documentation]    Set up all of the tests in this suite
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig
+    Upload String To File   ${GREENBOOT_CONFIG_CONTENT}    ${GREENBOOT_CONFIG_FILE} # change the default timeout
+
+Upload User Config
```

Comment:
The files being manipulated here aren't really "configuration" files. They're the workload and check script, right? This might be better named "Add Workload" or something along those lines. That would lead to changing the suite variable names, too.